compiler/rustc_codegen_cranelift/src/abi/mod.rs

```diff
@@ -51,6 +51,11 @@ pub(crate) fn conv_to_call_conv(
         CanonAbi::Rust | CanonAbi::C => default_call_conv,
         CanonAbi::RustCold => CallConv::Cold,
 
+        // Functions with this calling convention can only be called from assembly, but it is
+        // possible to declare an `extern "custom"` block, so the backend still needs a calling
+        // convention for declaring foreign functions.
+        CanonAbi::Custom => default_call_conv,
```

Comment:
This seems reasonable but I wonder if down the line clif (or the other backends) might want to have some kind of calling convention representation that can be `bug!`ed on if it gets directly called somewhere. (cc @bjorn3)